tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,139 @@
+        assert_script_run("virsh net-create vnet_host_bridge.xml");
+        assert_script_run("virsh attach-interface $guest network vnet_host_bridge --live");
+        my $mac_host_bridge = script_output("sleep 30;virsh domiflist $guest | grep vnet_host-bridge| grep -E -o '[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}'");
+        my $gi_host_bridge = script_output("arp | grep $mac_host_bridge| awk \'{print $1}\'");
+        assert_script_run("ssh root\@$gi_host_bridge 'ping -c5 -W1 -I eth1 10.160.0.207'",60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest bridge --mac $mac_host_bridge --current");
+        assert_script_run("virsh net-destroy vnet_host_bridge");
+
+    }
+
+#ISOLATED NETWORK
+#Prepare Test
+my $vnet_isolated_bridge = <<'EOF';
+<network>
+ <name>vnet_isolated_bridge</name>
+ <bridge name="virbr3" />
+ <ip address="192.168.152.1" netmask="255.255.255.0">                                                   
+  <dhcp>
+   <range start="192.168.152.2" end="192.168.152.254" />                                                
+  </dhcp>
+ </ip>
+ </network>
+EOF
+    foreach my $guest (keys %xen::guests) {
+        record_info "virsh-virtual_network", "BRIDGE - ISOLATED NETWORK";
+        assert_script_run("echo '$vnet_isolated_bridge' >> vnet_isolated_bridge.xml");
+        assert_script_run("virsh net-create vnet_isolated_bridge.xml");
+        assert_script_run("virsh attach-interface $guest network vnet_isolated_bridg --live");
+	#my $gi_vnet_isolated_bridge = script_output("sleep 30;journalctl | grep nsmasq-dhcp| tail -1| grep -oE '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'");
+        assert_script_run("ssh root\@$guest ip route");
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+        assert_script_run("virsh net-destroy vnet_isolated_bridge");
+    }
+
+#ROUTED NETWORK
+#Prepare Test
+my $vnet_routed = <<'EOF';
+<network>
+ <name>vnet_routed</name>
+ <bridge name="virbr2" />
+ <forward mode="route" dev="br0"/>1
+ <ip address="192.168.129.1" netmask="255.255.255.0">                                                   
+  <dhcp>
+   <range start="192.168.129.2" end="192.168.129.254" />                                                
+  </dhcp>
+ </ip>
+</network>
+EOF
+    foreach my $guest (keys %xen::guests) {
+        record_info "virsh-virtual_network.xml", "ROUTED NETWORK";                           
+        assert_script_run("echo '$vnet_routed' >> vnet_routed.xml"); 
+        assert_script_run("virsh net-define vnet_routed.xml");
+        assert_script_run("virsh net-start vnet_routed");
+        assert_script_run("virsh attach-interface $guest network vnet_routed --model virtio --live");
+	#my $gi_vnet_routed = script_output("sleep 30;journalctl | grep dnsmasq-dhcp| tail -1| grep -oE "([0-9]{1,3}[\.]){3}[0-9]{1,3}"");
+        assert_script_run("ssh root\@$guest 'ping -c5 -W1 -I eth1 10.160.0.207'",60); 
```

Comment:
Please use the same public server for connection testing from guest to public external network.

Reply:
Here, just try to use openqa.suse.de as the public serve for for connection testing from guest to public external network.